dgv/metrics/task_functions.py

```diff
@@ -126,329 +238,296 @@ def get_dict(df, obj_property):
-                slug, found, detect = search_pattern(patterns_reuses, item, "reuses")
-            if not found:
-                slug, found, detect = search_pattern(patterns_organizations, item, "organizations")
-            if not found:
-                slug, found, detect = search_pattern_resource_static(
-                    pattern_resources_static,
-                    item,
-                    "resources-static"
+def get_info(parsed_line: str) -> tuple[Optional[str], Optional[str], Optional[str]]:
+    """
+    Retrieve information related to the datasets, organisation or resources
+    from the anonymised HAProxy logs.
+
+    Args:
+        parsed_line (str): HAProxy line to parse
+
+    Example:
+        parsed_line = '2024-11-13T00:00:23.927326+01:00 slb-04 haproxy[260742]: 127.0.0.1:37959 '
+                      '[13/Nov/2024:00:00:23.908] DATAGOUVFR_RGS~ DATAGOUVFR_NEWINFRA/dataweb-06 '
+                      '0/0/2/16/+18 302 +684 - - --NN 222/189/4/1/0 0/0 '
+                      '"GET /fr/datasets/r/ee16d126-af0f-4b3b-84d3-080ef8bc0abd HTTP/1.1"'
+        Output:
+            slug_line = "ee16d126-af0f-4b3b-84d3-080ef8bc0abd"
+            type = "resources"
+            segment = "fr"
+    """
+    static_slug_line, static_obj_type, static_segment = (None, None, None)
+    # DATAGOUVFR_RGS: service behind the RGS certificate so www.data.gouv.fr
+    url_match = re.search(
+        r" DATAGOUVFR_RGS~ .* (200|302) .* \"GET (/[^\s]+)", parsed_line
+    )
+    if url_match:
+        id_match = url_match.group(2)
+        for obj_config in OBJ_CONFIG:
+            for segment, pattern in obj_config.log_patterns.items():
+                if pattern in id_match:
+                    if segment in OBJ_STATIC_LIST:
+                        # Lowest pattern priority
+                        static_slug_line = (
+                            f"https://static.data.gouv.fr{url_match}".replace(";", "")
+                        )
+                        static_obj_type = segment
+                        static_segment = segment
+                    else:
+                        slug_line = (
+                            id_match.replace(pattern, "").split("/")[0].replace(";", "")
+                        )
+                        return slug_line, obj_config.type, segment
+
+    return static_slug_line, static_obj_type, static_segment
+
+
+def save_list_obj(list_obj: list[dict[str, str]], list_of_types: list[str]) -> None:
+    """
+    Each dict in the input list must contain a 'type' key. The function categorizes
+    objects based on their types and writes them to corresponding CSV files.
+
+    Args:
+        list_obj (list[dict[str, str]]): A list of dictionaries where each dictionary represents
+            an object with keys such as 'resources', 'datasets', etc.
+        list_of_types (list[str]): The list of type keys.
+    """
+    # Split the objects per type
+    lists_per_type = {type: [] for type in list_of_types}
```

Comment:
I recently learnt about `collections.defaultdict` which would be a perfect fit here :eyes:
```suggestion
    lists_per_type = defaultdict(list)
```

Reply:
TIL! Thanks!